backpack-compose/src/main/kotlin/net/skyscanner/backpack/compose/bottomsheet/BpkBottomSheet.kt

```diff
@@ -19,147 +19,97 @@
-import androidx.compose.material.ExperimentalMaterialApi
-import androidx.compose.material.Surface
+import androidx.compose.material3.BottomSheetScaffold
+import androidx.compose.material3.ExperimentalMaterial3Api
+import androidx.compose.material3.rememberBottomSheetScaffoldState
 import androidx.compose.runtime.Composable
 import androidx.compose.runtime.getValue
 import androidx.compose.runtime.mutableStateOf
 import androidx.compose.runtime.remember
-import androidx.compose.runtime.rememberCoroutineScope
 import androidx.compose.runtime.setValue
 import androidx.compose.ui.Alignment
 import androidx.compose.ui.Modifier
-import androidx.compose.ui.composed
 import androidx.compose.ui.draw.alpha
 import androidx.compose.ui.layout.onGloballyPositioned
 import androidx.compose.ui.platform.LocalDensity
-import androidx.compose.ui.semantics.collapse
-import androidx.compose.ui.semantics.expand
-import androidx.compose.ui.semantics.semantics
 import androidx.compose.ui.unit.Dp
-import androidx.compose.ui.unit.IntOffset
 import androidx.compose.ui.unit.dp
-import kotlinx.coroutines.launch
 import net.skyscanner.backpack.compose.theme.BpkTheme
 import net.skyscanner.backpack.compose.tokens.BpkBorderRadius
 import net.skyscanner.backpack.compose.tokens.BpkElevation
 import net.skyscanner.backpack.compose.tokens.BpkSpacing
-import net.skyscanner.backpack.compose.utils.nestedScrollFixedSwipeable
-import kotlin.math.roundToInt
 
-@OptIn(ExperimentalMaterialApi::class)
+@OptIn(ExperimentalMaterial3Api::class)
 @Composable
 fun BpkBottomSheet(
     sheetContent: @Composable ColumnScope.(PaddingValues) -> Unit,
     modifier: Modifier = Modifier,
-    state: BpkBottomSheetState = rememberBpkBottomSheetState(BpkBottomSheetValue.Collapsed),
+    state: BpkBottomSheetState = rememberBpkBottomSheetState(),
     sheetGesturesEnabled: Boolean = true,
     peekHeight: Dp = BottomSheetScaffoldDefaults.SheetPeekHeight,
     content: @Composable (PaddingValues) -> Unit,
 ) {
+    var contentHeight by remember { mutableStateOf(0) }
+    val totalSheetHeight = peekHeight + HandleHeight
 
-    BoxWithConstraints(modifier) {
-        val fullHeight = constraints.maxHeight.toFloat()
-        val peekHeightWithHandle = peekHeight + HandleHeight
-        val peekHeightPx = with(LocalDensity.current) { peekHeightWithHandle.toPx() }
-        var bottomSheetHeight by remember { mutableStateOf(fullHeight) }
-        val progress = state.wrapped.progress
-        val openingPercent = when (progress.to) {
-            BpkBottomSheetValue.Expanded -> (1f - progress.fraction)
-            BpkBottomSheetValue.Collapsed -> progress.fraction
+    var openingPercent = if (state.currentValue == BpkBottomSheetValue.Collapsed) 1f else 0f
+    if (contentHeight > 0) {
+        val total = contentHeight - with(LocalDensity.current) { totalSheetHeight.toPx() }
+        val currentPosition = try {
+            state.delegate.requireOffset()
```

Comment:
This method throws an exception if being called before the first layout pass.
We bypass it by calling it only when `contentHeight` is greater than zero (which only gets updated after 1st layout pass).

However, I put this try/catch block here just in case – we don't want to crash accidentally.